react/training/src/redux/login/actions.js

```diff
@@ -0,0 +1,37 @@
+import { getByEmail } from '../../services/loginService';
+
+export const actionTypes = {
+  LOG_SUCCESS: 'LOG_SUCCESS',
+  LOG_FAILURE: 'LOG_FAILURE',
+  LOG_REQUEST: 'LOG_REQUEST'
+};
+
+export const loginActions = {
+  handleSubmit: userLogged => async dispatch => {
+    dispatch({ type: actionTypes.LOG_REQUEST, payload: true });
+    const response = await getByEmail(userLogged);
+    if (response.status === 200) {
+      if (response.data.length > 0) {
+        const authenticated = response.data[0].password === userLogged.password;
+        if (authenticated) {
+          dispatch({
+            type: actionTypes.LOG_SUCCESS,
+            payload: authenticated
+          });
+        } else {
+          dispatch({
+            type: actionTypes.LOG_FAILURE,
+            payload: 'Error: The password is not correct'
+          });
+        }
+      } else {
+        dispatch({
```

Comment:
You dont need to dispatch `LOG_FAILURE` twice. Just ask `response.data.length > 0 && responde.data[0].password ...`

Reply:
`response.status === 200` is not necessary. This is perfect, but not necessary. You can just ask for `response.data`.